### PR TITLE
executor: support selecting store by tidb_replica_read for stale read

### PR DESCRIPTION
**Pull request by Yisaer**
### What is changed and how it works?
Add an extra value choice `closest-replicas` for `tidb_replica_read` and make `follower read` and `stale read`  can be affected by `tidb_replica_read` with same behavior.

### Check List 

Tests 

- [x] Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: executor/stale_txn_test.go [hunk 1]
```diff
@@ -244,6 +244,7 @@ func (s *testStaleTxnSerialSuite) TestSelectAsOf(c *C) {
 }
 
 func (s *testStaleTxnSerialSuite) TestStaleReadKVRequest(c *C) {
+	//c.Skip("tmp skip")
```
Comment (JmPotato) [inline executor/stale_txn_test.go:247]: Why commenting this line?

Comment (xhebox) [reply executor/stale_txn_test.go:247]: It seems useless.

**Review by xhebox — APPROVED** [review]
LGTM

File: executor/stale_txn_test.go (hunk 1 above)
Comment (Yisaer) [reply executor/stale_txn_test.go]: removed.

**Review by JmPotato — APPROVED** [review]

**Pull request merged**